ledger/src/blockstore_processor.rs

```diff
@@ -2275,6 +2283,38 @@ pub fn cache_block_meta(bank: &Arc<Bank>, cache_block_meta_sender: Option<&Cache
     }
 }
 
+pub type RewardsBatch = (Slot, KeyedRewardsAndNumPartitions);
+pub enum RewardsMessage {
+    Batch(RewardsBatch),
+    Complete(Slot),
+}
+
+#[derive(Clone, Debug)]
+pub struct RewardsRecorderSender {
+    pub sender: Sender<RewardsMessage>,
+}
+pub type RewardsRecorderReceiver = Receiver<RewardsMessage>;
```

Comment:
what. why not just type alias both?

Reply:
The `Sender` is not a simple alias so that I could avoid duplicating this piece in the callsites (`ReplayStage` and `ledger-tool`):
```rust
impl RewardsRecorderSender {
    pub fn send_rewards(&self, bank: &Bank) {
        let rewards = bank.get_rewards_and_num_partitions();
        if rewards.should_record() {
            self.sender
                .send(RewardsMessage::Batch((bank.slot(), rewards)))
                .unwrap_or_else(|err| warn!("rewards_recorder_sender failed: {:?}", err));
        }
        self.sender
            .send(RewardsMessage::Complete(bank.slot()))
            .unwrap_or_else(|err| warn!("rewards_recorder_sender failed: {:?}", err));
    }
}
```
Not a ton of actual duplicate code I know, but was following the existing pattern for `TransactionStatusSender`:
https://github.com/anza-xyz/agave/blob/397e928494c6451fc27346ef9cff4dee414a207f/ledger/src/blockstore_processor.rs#L2227-L2228
Given the extra fetching of rewards + `should_record()`, I'd say `RewardsRecorderSender` has better justification for custom-impl-instead-of-type-alias than `TransactionStatusSender`, but I also don't feel too strongly about this one